test/bats/test.bats

```diff
@@ -0,0 +1,176 @@
+#!/usr/bin/env bats
+
+load helpers
+
+BATS_TESTS_DIR=test/bats/tests
+WAIT_TIME=120
+SLEEP_TIME=1
+CLEAN_CMD="echo cleaning..."
+
+teardown() {
+  bash -c "${CLEAN_CMD}"
+}
+
+@test "gatekeeper-controller-manager is running" {
+  wait_for_process ${WAIT_TIME} ${SLEEP_TIME} "kubectl -n gatekeeper-system wait --for=condition=Ready --timeout=60s pod -l control-plane=controller-manager"
+}
+
+@test "gatekeeper-audit is running" {
+  wait_for_process ${WAIT_TIME} ${SLEEP_TIME} "kubectl -n gatekeeper-system wait --for=condition=Ready --timeout=60s pod -l control-plane=audit-controller"
+}
+
+@test "namespace label webhook is serving" {
+  cert=$(mktemp)
+  CLEAN_CMD="${CLEAN_CMD}; rm ${cert}"
+  wait_for_process ${WAIT_TIME} ${SLEEP_TIME} "get_ca_cert ${cert}"
+
+  kubectl run temp --generator=run-pod/v1  --image=tutum/curl -- tail -f /dev/null
```

Comment:
How come this was needed? Could we add a comment explaining? I think we should also look to make this change in upstream Gatekeeper.

Reply:
This was a straight port from upstream

Reply:
Ah, so they may have fixed those issues we were hitting.